tests/models/whisper/test_modeling_tf_whisper.py

```diff
@@ -804,6 +809,192 @@ def _test_large_batched_generation(in_queue, out_queue, timeout):
+        return config, inputs_dict
+
+    def get_subsampled_output_lengths(self, input_lengths):
+        """
+        Computes the output length of the convolutional layers
+        """
+
+        for i in range(self.num_conv_layers):
+            input_lengths = (input_lengths - 1) // 2 + 1
+
+        return input_lengths
+
+    @property
+    def encoder_seq_length(self):
+        return self.get_subsampled_output_lengths(self.seq_length)
+
+    def create_and_check_model_forward(self, config, inputs_dict, freeze_encoder=False):
+        model = TFWhisperForAudioClassification(config=config)
+
+        if freeze_encoder:
+            model.freeze_encoder()
+
+        input_features = inputs_dict["input_features"]
+
+        # first forward pass
+        last_hidden_state = model(input_features).logits
+
+        self.parent.assertTrue(last_hidden_state.shape, (13, 2))
+
+
+def prepare_whisper_encoder_inputs_dict(config, input_features, head_mask=None):
+    if head_mask is None:
+        head_mask = tf.ones([config.encoder_layers, config.encoder_attention_heads])
+    return {"input_features": input_features, "head_mask": head_mask}
+
+
+@require_tf
+class TFWhisperEncoderModelTest(TFModelTesterMixin, unittest.TestCase):
+    all_model_classes = (TFWhisperForAudioClassification,) if is_tf_available() else ()
+    is_encoder_decoder = False
+    fx_compatible = False
+    test_pruning = False
+    test_missing_keys = False
+
+    input_name = "input_features"
+
+    def setUp(self):
+        self.model_tester = TFWhisperEncoderModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=WhisperConfig)
+        self.maxDiff = 3000
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_forward_signature(self):
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            signature = inspect.signature(model.call)
+            # signature.parameters is an OrderedDict => so arg_names order is deterministic
+            arg_names = [*signature.parameters.keys()]
+
+            expected_arg_names = ["input_features", "head_mask", "encoder_outputs"]
+            self.assertListEqual(arg_names[: len(expected_arg_names)], expected_arg_names)
+
+    # input embeds is meaningless for an encoder-only acoustic model
+    def test_inputs_embeds(self):
+        pass
+
+    # the equivalent test is passing the encoder outputs directly to the model
+    def test_encoder_outputs(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+
+            inputs = copy.deepcopy(self._prepare_for_class(inputs_dict, model_class))
+            
+            with tf.stop_gradient:
+                outputs = model(**inputs)[0]
```

Comment:
You can simply do:
```suggestion
            outputs = model(**inputs)[0]
```
As we do for the encoder-decoder standalone test previously:
https://github.com/huggingface/transformers/blob/8f76dc8e5aaad58f2df7748b6d6970376f315a9a/tests/models/whisper/test_modeling_tf_whisper.py#L238

Stop gradient is used to stop the gradient back propagating through our graph when training - you typically call it on a tensor that you want to freeze or not back prop through (_c.f._ https://www.tensorflow.org/api_docs/python/tf/stop_gradient)

Reply:
By applying just this change, there was an error that was thrown about always passing the first argument to `Layer.call()`. I think it has something to do with this line which I had duplicated in the TF tests:
https://github.com/huggingface/transformers/blob/8f76dc8e5aaad58f2df7748b6d6970376f315a9a/tests/models/whisper/test_modeling_whisper.py#L1614